pkg/operator/encryption/state.go

```diff
@@ -26,13 +26,18 @@ func getCurrentEncryptionConfig(secrets corev1client.SecretInterface, revision s
 	if err != nil {
 		// if encryption is not enabled at this revision or the secret was deleted, we should not error
 		if errors.IsNotFound(err) {
-			return &apiserverconfigv1.EncryptionConfiguration{}, nil
+			return nil, nil
 		}
 		return nil, err
 	}
 
+	data, ok := encryptionConfigSecret.Data[encryptionConfSecret]
+	if !ok {
+		return nil, nil
```

Comment:
does it mean that the current encryption config is broken?
shouldn't we report an error in this case?

Reply:
We just behave like for a missing config. We can recover.